Source/HelixToolkit.SharpDX.Shared/Model/Scene/CrossSectionMeshNode.cs

```diff
@@ -238,6 +240,86 @@ protected override RenderCore OnCreateRenderCore()
             {
                 return new CrossSectionMeshRenderCore();
             }
+
+            protected override bool OnHitTest(RenderContext context, Matrix totalModelMatrix, ref Ray rayWS, ref List<HitTestResult> hits)
+            {
+                int hitsBeforeCheck = hits?.Count ?? 0;
+                var meshGeometry3d = Geometry as MeshGeometry3D;
+                if (meshGeometry3d == null)
```

Comment:
Is it correct to return false or is it better to cast (and get an exception) or at least to log this?

Reply:
Just return false I think. If it is not MeshGeometry3D, the mesh will not be able to render on screen.